ext/bcmath/libbcmath/src/div.c

```diff
@@ -31,219 +31,446 @@
+	size_t numerator_arr_size = (numerator_readable_len + numerator_bottom_extension + BC_VECTOR_SIZE - 1) / BC_VECTOR_SIZE;
+	size_t quot_arr_size = numerator_arr_size - divisor_arr_size + 1;
+	size_t quot_real_arr_size = MIN(quot_arr_size, (quot_len + BC_VECTOR_SIZE - 1) / BC_VECTOR_SIZE);
+
+	BC_VECTOR *numerator_vectors = safe_emalloc(numerator_arr_size + divisor_arr_size + quot_arr_size, sizeof(BC_VECTOR), 0);
+	BC_VECTOR *divisor_vectors = numerator_vectors + numerator_arr_size;
+	BC_VECTOR *quot_vectors = divisor_vectors + divisor_arr_size;
+
+	/* Fill with zeros and convert as many vector elements as needed */
+	size_t numerator_vector_count = 0;
+	while (numerator_bottom_extension >= BC_VECTOR_SIZE) {
+		numerator_vectors[numerator_vector_count] = 0;
+		numerator_bottom_extension -= BC_VECTOR_SIZE;
+		numerator_vector_count++;
+	}
 
-/* The full division routine. This computes N1 / N2.  It returns
-   true if the division is ok and the result is in QUOT.  The number of
-   digits after the decimal point is SCALE. It returns false if division
-   by zero is tried. The algorithm is found in Knuth Vol 2. p237. */
-bool bc_divide(bc_num n1, bc_num n2, bc_num *quot, int scale)
+	size_t numerator_bottom_read_len = BC_VECTOR_SIZE - numerator_bottom_extension;
+
+	size_t base;
+	size_t numerator_read = 0;
+	if (numerator_bottom_read_len < BC_VECTOR_SIZE) {
+		numerator_read = MIN(numerator_bottom_read_len, numerator_readable_len);
+		base = POW_10_LUT[numerator_bottom_extension];
+		numerator_vectors[numerator_vector_count] = 0;
+		for (size_t i = 0; i < numerator_read; i++) {
+			numerator_vectors[numerator_vector_count] += *numerator * base;
+			base *= BASE;
+			numerator--;
+		}
+		numerator_vector_count++;
+	}
+
+	/* Bulk convert numerator and divisor to vectors */
+	if (numerator_readable_len > numerator_read) {
+		bc_convert_to_vector(numerator_vectors + numerator_vector_count, numerator, numerator_readable_len - numerator_read);
+	}
+	bc_convert_to_vector(divisor_vectors, divisor, divisor_len);
+
+	/* Do the divide */
+	if (divisor_arr_size == 1) {
+		bc_fast_div(numerator_vectors, numerator_arr_size, divisor_vectors[0], quot_vectors, quot_arr_size);
+	} else {
+		bc_standard_div(numerator_vectors, numerator_arr_size, divisor_vectors, divisor_arr_size, divisor_len, quot_vectors, quot_arr_size);
+	}
+
+	/* Convert to bc_num */
+	char *qptr = (*quot)->n_value;
+	char *qend = qptr + quot_len - 1;
+
+	size_t i;
+	for (i = 0; i < quot_real_arr_size - 1; i++) {
+#if BC_VECTOR_SIZE == 4
+		bc_write_bcd_representation(quot_vectors[i], qend - 3);
+		qend -= 4;
+#else
+		bc_write_bcd_representation(quot_vectors[i] / 10000, qend - 7);
+		bc_write_bcd_representation(quot_vectors[i] % 10000, qend - 3);
+		qend -= 8;
+#endif
+	}
+
+	while (qend >= qptr) {
+		*qend-- = quot_vectors[i] % BASE;
+		quot_vectors[i] /= BASE;
+	}
+
+	efree(numerator_vectors);
+}
+
+bool bc_divide(bc_num n1, bc_num n2, bc_num *quot, size_t scale)
```

Comment:
Once again renaming the n1 and n2 variables here would help.

Reply:
Fixed 
https://github.com/php/php-src/pull/14660/commits/68bcff9eb855af39c3efa19f6cc0e41c8263f5d4